hw_submission_automation/hw_submission_automation.py

```diff
@@ -402,54 +479,19 @@ def gen_guest_mapping():
     return mapping
 
 
-def main(argv):
-    test_harness = "HLK"
-    product_name = None
-    guest_names = []
-    guest_arch = "x64"
-    submission_name = None
-    package_path = None
-    announcement_date = "2025-01-01"
+def main_submit(args):
+    # Validate required arguments for submit action
+    if not args.product_name:
+        print("Error: --product_name is required for submit action")
```

Comment:
Consider using **argparse.ArgumentParser.error()** for consistent error reporting.